src/main/java/slimeknights/mantle/fluid/attributes/FluidAttributes.java

```diff
@@ -0,0 +1,443 @@
+package slimeknights.mantle.fluid.attributes;
+
+import javax.annotation.Nullable;
+
+import io.github.fabricators_of_create.porting_lib.util.FluidStack;
+import net.minecraft.core.Registry;
+import net.minecraft.world.item.BucketItem;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.dimension.DimensionType;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.level.BlockAndTintGetter;
+
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.item.Rarity;
+import net.minecraft.client.renderer.BiomeColors;
+
+import net.minecraft.Util;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.sounds.SoundEvent;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.phys.BlockHitResult;
+
+/**
+ * Minecraft Forge Fluid Implementation
+ *
+ * This class is a fluid (liquid or gas) equivalent to "Item." It describes the nature of a fluid
+ * and contains its general properties.
+ *
+ * These properties do not have inherent gameplay mechanics - they are provided so that mods may
+ * choose to take advantage of them.
+ *
+ * Fluid implementations are not required to actively use these properties, nor are objects
+ * interfacing with fluids required to make use of them, but it is encouraged.
+ *
+ * The default values can be used as a reference point for mods adding fluids such as oil or heavy
+ * water.
+ *
+ */
+public class FluidAttributes
+{
+  public static final int BUCKET_VOLUME = 1000;
```

Comment:
This should point to FluidConstants.BUCKET. It seems I actually forgot to do this in my 1.19.4 port